File: greedy/partitionStringIntoSubstringsWithValuesAtMostK.py
# https://leetcode.com/problems/partition-string-into-substrings-with-values-at-most-k/

class Solution(object):
    def minimumPartition(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        number = 0
        result =0
        
        for num in s:
            
            number = number * 10 + int(num)

            if number > k:
                number = int(num)
                result +=1
                
            if number > k:
                return -1

        return result+1

File: greedy/partitionStringIntoSubstringsWithValuesAtMostK.java
package greedy;

class partitionStringIntoSubstringsWithValuesAtMostK {
    public int minimumPartition(String s, int k) {
        
        long number = 0 ;
        int n = s.length();
        int result = 0;

        for (int i = 0 ; i< n; i++){
            number = number * 10 + (s.charAt(i)-'0');
            if (number > k){
                number = s.charAt(i)-'0';
                result +=1;
            }
            if (number > k){
                return -1;
            }
        }
        return result+1;
    }
}
